### DISCUSSION TITLE
Welcome to opensnitch Discussions!

### DISCUSSION THREAD

**Discussion by evilsocket** [evilsocket/opensnitch#348] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by ghost** [comment]
thanks! this seems like something i wanted to have on linux since i stared using it 20 years ago.
what i would like is more detailed instructions on compiling the sources - dependencies and such, i tried that by following the wiki article and failed (having go issues)

**Comment by themighty1** [comment]
Hi @NESC1US , compilation instructions work for me. Make sure you have Go >= 1.13 
Feel free to post here your error log, we'll take a look at it.

**Comment by Herve5** [comment]
I am an ever more naive end-user than the ones above : I just download executables, since the moment they have been available here. This works absolutely fine, on Debian at least. 
All the best, Evilsocket!!
H.

**Reply by evilsocket** [reply]
thanks! it's mostly thanks to @gustavo-iniguez-goya and [other cotributors](https://github.com/evilsocket/opensnitch/graphs/contributors) who did an amazing job, i just kickstarted the first versions of the project :D

**Comment by themighty1** [comment]
@NESC1US , the compilation instructions were outdated. This may have caused your issues. We've updated then. Give it a try https://github.com/evilsocket/opensnitch/wiki/Compilation

**Reply by bgoodwin91006-1** [reply]
Hi all,
I just followed the instructions on the Compilation page and the compile failed. The only change I made to the instructions was the GOPATH. My GOPATH is set to a 'go' directory in my home directory due to previously compiling lxd (the container software). Here is the terminal output from issuing the make command;

```
make[1]: Entering directory '/home/bob/opensnitch/proto'
protoc -I. ui.proto --go_out=../daemon/ui/protocol/ --go-grpc_out=../daemon/ui/protocol/ --go_opt=paths=source_relative --go-grpc_opt=paths=source_relative
protoc-gen-go: program not found or is not executable
--go_out: protoc-gen-go: Plugin failed with status code 1.
make[1]: *** [Makefile:4: ../daemon/ui/protocol/ui.pb.go] Error 1
make[1]: Leaving directory '/home/bob/opensnitch/proto'
make: *** [Makefile:8: protocol] Error 2

```

**Reply by ghost** [reply]
Your problem is described on 3rd line - check if a directory containing protoc-gen-go is in $PATH and if the binary is executable.

**Reply by bgoodwin91006-1** [reply]
Hi NES1US,
Thank you for that amazingly fast response. Also, thank you for this excellent program. I was thinking of doing something like this myself a year ago as I started learning about bpf. I have wanted an app filter for the network since I switched to Linux many years ago. I did install the binary but I would still like to do the compile. The following is my PATH, GOPATH, and protoc-gen-go permissions.

$ echo $PATH (go references (2) are at the end of the list)
/home/bob/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/home/bob/go:/home/bob/go/bin

$ ls -l /home/bob/go/bin/protoc-gen-go
-rwxrwxr-x 1 bob bob 8243122 Oct 27 21:36 /home/bob/go/bin/protoc-gen-go

$ echo $GOPATH
/home/bob/go

As you can see, my PATH env still reflects the go and go/bin directories from when I compiled lxd. The permissions for protoc-gen-go look right. I don't understand why I'm getting this error.

I took a quick look at the opensnitch and  opensnitch/proto makefiles but I confess I am unfamiliar with a lot of the go type references. Altogether, this exercise has got me interested learning go.

Best regards,
Bob G.

**Comment by ski007** [comment]
Thank you for the great program!
the number one firewall for Linux :-)

 **@gustavo-iniguez-goya** and **@evilsocket**  thank you very much for working on this program
